sha2/src/lib.rs

```diff
@@ -72,15 +73,50 @@ pub use sha512::compress512;
 
 pub use core_api::{Sha256VarCore, Sha512VarCore};
 
+/// SHA-224 core.
+pub type Sha224Core = CtVariableCoreWrapper<Sha256VarCore, U28>;
 /// SHA-224 hasher.
-pub type Sha224 = CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, U28>>;
+pub type Sha224 = CoreWrapper<Sha224Core>;
+/// SHA-224 core serialized state.
+pub type Sha224CoreSerializedState =
+    GenericArray<u8, <Sha224Core as HashCoreSerializableState>::SerializedStateSize>;
+
+/// SHA-256 core.
+pub type Sha256Core = CtVariableCoreWrapper<Sha256VarCore, U32>;
 /// SHA-256 hasher.
-pub type Sha256 = CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, U32>>;
-/// SHA-512/224 hasher.
-pub type Sha512_224 = CoreWrapper<CtVariableCoreWrapper<Sha512VarCore, U28>>;
-/// SHA-512/256 hasher.
-pub type Sha512_256 = CoreWrapper<CtVariableCoreWrapper<Sha512VarCore, U32>>;
+pub type Sha256 = CoreWrapper<Sha256Core>;
+/// SHA-256 core serialized state.
+pub type Sha256CoreSerializedState =
+    GenericArray<u8, <Sha256Core as HashCoreSerializableState>::SerializedStateSize>;
+
+/// SHA-512/224 core.
+pub type Sha512_224Core = CtVariableCoreWrapper<Sha512VarCore, U28>;
+/// SHA-512/224 core.
+pub type Sha512_224 = CoreWrapper<Sha512_224Core>;
+/// SHA-512/224 core serialized state.
+pub type Sha512_224CoreSerializedState =
+    GenericArray<u8, <Sha512_224Core as HashCoreSerializableState>::SerializedStateSize>;
```

Comment:
I don't think we need to introduce a whole bunch of type aliases for serialized states.